recipe/mpiexec.sh

```diff
@@ -1,14 +1,16 @@
 #!/bin/bash
 set -euo pipefail
 
-mpiexec="mpiexec"
-if [[ "$mpi" == "mpich" ]]; then
+if command -v "${PREFIX}/bin/mpichversion" >/dev/null; then
     export HYDRA_LAUNCHER=fork
-elif [[ "$mpi" == "openmpi" ]]; then
+    mpiexec="mpiexec"
+fi
+
+if command -v "${PREFIX}/bin/ompi_info" >/dev/null; then
     export OMPI_MCA_plm=isolated
     export OMPI_MCA_rmaps_base_oversubscribe=yes
+    export OMPI_MCA_btl=sm,self
```

Comment:
```suggestion
    export OMPI_MCA_btl=vader,self
```

Current error suggests to not use `sm`.